ReserveRoom.java

```diff
@@ -3,106 +3,93 @@
 import java.util.Scanner;
 
 public class ReserveRoom {
-    private List<Room> rooms; // List of rooms
-    private List<String> reservations; // List of room reservations
+    private List<Room> rooms;         
+    private List<String> reservations;  
 
-    // Constructor initializes the list of rooms and reservations
     public ReserveRoom(List<Room> rooms) {
         this.rooms = rooms;
         this.reservations = new ArrayList<>();
     }
+    private User promptUserDetails(Scanner scanner) {
+        System.out.print("Enter your name: ");
+        String name = scanner.nextLine();
+        System.out.print("Enter your email: ");
+        String email = scanner.nextLine();
+        return new User(name, email);
+    }
 
-    // Method to reserve a room for a user
     public String reserveRoom(User user, Room room) {
-        // Check if the room is available
         if (!room.isAvailable()) {
             return "Room " + room.getNumber() + " is already reserved.";
         }
-        // Reserve the room
-        room.setAvailable(false); // Mark the room as unavailable
+        room.setAvailable(false);
         reservations.add("User " + user.getName() + " reserved room " + room.getNumber());
-
         return "Room " + room.getNumber() + " reserved successfully for " + user.getName() + ".";
     }
-    // Method to view available rooms
+
     public void displayAvailableRooms() {
-        boolean foundAvailableRoom = false;
+        boolean found = false;
```

Comment:
Replace the variable found in the displayAvailableRooms method with the more descriptive name hasAvailableRooms. And update all references inside displayAvailableRooms() accordingly. (Readability)
```suggestion
        boolean hasAvailableRooms = false;
```